src/join.rs

```diff
@@ -1,6 +1,11 @@
 use std;
+use std::cell::UnsafeCell;
+
+use hibitset::{BitSetAnd, BitIter, BitSetLike, BitProducer};
+
+use rayon::iter::ParallelIterator;
+use rayon::iter::internal::{UnindexedProducer, UnindexedConsumer, Folder, bridge_unindexed};
```

Comment:
Doesn't really matter because it's readable both ways; I usually do place the imports which access the deeper hierarchy at the end.

But no need to spend too much time on that, I'll do a cleanup PR after this one is merged anyways.